lib/ontologies_linked_data/models/mod/semantic_artefact_catalog_record.rb

```diff
@@ -0,0 +1,112 @@
+      attribute_mapped :ontologyType, mapped_to: { model: :ontology }
+      attribute_mapped :classType, mapped_to: { model: :ontology_submission }
+      attribute_mapped :missingImports, mapped_to: { model: :ontology_submission }
+      attribute_mapped :submissionStatus, mapped_to: { model: :ontology_submission }
+      attribute_mapped :pullLocation, mapped_to: { model: :ontology_submission }
+      attribute_mapped :dataDump, namespace: :void, mapped_to: { model: :ontology_submission }
+      attribute_mapped :csvDump, mapped_to: { model: :ontology_submission }
+      attribute_mapped :uploadFilePath, mapped_to: { model: :ontology_submission }
+      attribute_mapped :diffFilePath, mapped_to: { model: :ontology_submission }
+      attribute_mapped :masterFileName, mapped_to: { model: :ontology_submission }
+
+
+      attribute :ontology, type: :ontology, enforce: [:existence]
+      attribute :submission, type: :ontology_submission
+
+      # Access control
+      read_restriction_based_on ->(record) { record.ontology }
+
+      serialize_default :acronym, :relatedArtefactId, :homepage, :created, :modified, :curatedOn, :curatedBy
+      serialize_never :ontology, :submission
+
+      def self.record_id_generator(record)
+        record.ontology.bring(:acronym) if !record.ontology.loaded_attributes.include?(:acronym)
+        raise ArgumentError, "Acronym is nil for ontology  #{record.ontology.id} to generate id" if record.ontology.acronym.nil?
+        return RDF::URI.new(
+          "#{(Goo.id_prefix)}records/#{CGI.escape(record.ontology.acronym.to_s)}"
+        )
+      end
+
+      ##
+      ## find an artefact (ontology) and map it to record
+      def self.find(artefact_id)
+        ont = Ontology.find(artefact_id).include(:acronym, :viewingRestriction, :administeredBy, :acl).first
+        return nil unless ont
+        new.tap do |sacr|
+          sacr.ontology = ont
+        end
+      end
+      
+      def self.all(attributes, page, pagesize)
+        all_count = Ontology.where.count
+        onts = Ontology.where.include(:acronym, :viewingRestriction, :administeredBy, :acl).page(page, pagesize).page_count_set(all_count).all
+        all_records = onts.map do |o|
+          new.tap do |sacr|
+            sacr.ontology = o
+            sacr.bring(*attributes) if attributes
+          end
+        end
+        Goo::Base::Page.new(page, pagesize, all_count, all_records)
+      end
+      
+      ##
+      ## find all records of one artefact
+      ## these are mapped to the submissions of an ontology
+      def artefact_all_records(attributes)
+        @ontology.bring(:submissions) unless @ontology.loaded_attributes.include?(:submissions)
+        all_records = @ontology.submissions.map do |s|
+          SemanticArtefactCatalogRecord.new.tap do |sacr|  
+            sacr.ontology = @ontology
+            sacr.submission = s
+            sacr.bring(*attributes) if attributes
```

Comment:
this will be really slow,  do the bring with the attributes in line 80.

Reply:
i'll remove `artefact_all_records` becuase it will return the submissions as record but we don't need it as we discussed last time.

Reply:
ok